packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -334,48 +356,65 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
         // if this domain has any Ontology Lookup field(s), check if we need to update the related field properties
         // based on the updated domain (i.e. check for any name changes to selected fields)
         // note: we skip any rowIndexChange which has a newIndex as those are just reorder changes
-        if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
-            ontologyLookupIndices.forEach(index => {
-                // skip any ontology lookup fields if they were removed
-                const ontFieldRemoved = rowIndexChange?.originalIndex === index;
-
-                if (!ontFieldRemoved) {
-                    updatedDomain = updateOntologyFieldProperties(
-                        // check for a field removal prior to the ontology lookup field
-                        rowIndexChange?.originalIndex < index ? index - 1 : index,
-                        domainIndex,
-                        updatedDomain,
-                        domain,
-                        rowIndexChange?.originalIndex
-                    );
+        if (rowIndexChanges) {
+            rowIndexChanges.forEach((rowIndexChange) => {
+                if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
+                    ontologyLookupIndices.forEach(index => {
+                        // skip any ontology lookup fields if they were removed
+                        const ontFieldRemoved = rowIndexChange?.originalIndex === index;
+
+                        if (!ontFieldRemoved) {
+                            updatedDomain = updateOntologyFieldProperties(
+                                // check for a field removal prior to the ontology lookup field
+                                rowIndexChange?.originalIndex < index ? index - 1 : index,
+                                domainIndex,
+                                updatedDomain,
+                                domain,
+                                rowIndexChange?.originalIndex
+                            );
+                        }
+                    });
                 }
             });
         }
 
-        this.setState(() => ({ fieldDetails: updatedDomain.getFieldDetails() }));
+        const visibleFieldsCount = getVisibleFieldCount(updatedDomain);
```

Comment:
From what I can see in the code all of the relevant callers to this `onDomainChange` function are already updating the visible field count/selection state, so I think this is redundant to set `visibleFieldsCount` here.

Reply:
True—tested this as well, removed